internal/service/remotecfg/remotecfg.go

```diff
@@ -260,24 +270,45 @@ var _ service.Service = (*Service)(nil)
 // run until the provided context is canceled or there is a fatal error.
 func (s *Service) Run(ctx context.Context, host service.Host) error {
 	s.ctrl = host.NewController(ServiceName)
+	defer func() {
+		s.mut.Lock()
+		if s.ticker != nil {
+			s.ticker.Stop()
+		}
+		s.mut.Unlock()
+	}()
 
 	s.fetch()
-	s.registerCollector()
+	err := s.registerCollector()
+	if err != nil {
+		return err
+	}
 
 	// Run the service's own controller.
 	go func() {
 		s.ctrl.Run(ctx)
 	}()
 
 	for {
+		// If the ticker is nil just wait, allow an update to set it.
```

Comment:
Why not set the ticker in the Run function? It also feels very random to wait 100 ms.

Reply:
I ended up iterating a few times on how to fix issues with the Ticker. When I added error checking for registerCollector() it caused ticker.Stop() to be called zero times (leaking goroutines) then twice (panic for closing closed channels). I ended up leaning in this direction as it was clearest and avoided the current behavior of having your tick time be at maxInt-jitter if the ticker wasn't configured.

I'll see if there's a better way to just initialize it in Run().

Reply:
I cleaned up the ticker code, should be better now & doesn't do the random wait.